src/mame/rm/rm380z.h

```diff
@@ -89,16 +95,23 @@ class rm380z_state : public driver_device
 	static inline constexpr int RM380Z_SCREENCOLS = 80;
 	static inline constexpr int RM380Z_SCREENROWS = 24;
 
+	// the HRG reference manual only mentions 12 memory pages for the display, but RML Extended BASIC v 5.0L
+	// writes beyond this to store character data for the HRG Level 2 "STPLOT" function.  Hence 16 pages are
+	// reserved here to support HRG text characters.
+	static inline constexpr int RM380Z_HRG_RAM_PAGES = 16;
+	static inline constexpr int RM380Z_HRG_RAM_SIZE = RM380Z_HRG_RAM_PAGES * 1280;
+	static inline constexpr int RM380Z_HRG_SCRATCHPAD_SIZE = 16;
+
 	bool ports_enabled_high() const { return ( m_port0 & 0x80 ); }
 	bool ports_enabled_low() const { return !( m_port0 & 0x80 ); }
 
 	bool get_rowcol_from_offset(int &row, int &col, offs_t offset) const;
 	void put_point(int charnum, int x, int y, int col);
 	void init_graphic_chars();
 
-	void putChar_vdu80(int charnum, int attribs, int x, int y, bitmap_ind16 &bitmap);
-	void putChar_vdu40(int charnum, int x, int y, bitmap_ind16 &bitmap);
-	void decode_videoram_char(int row, int col, uint8_t &chr, uint8_t &attrib);
+	void putChar_vdu80(int charnum, int attribs, int x, int y, bitmap_ind16 &bitmap) const;
+	void putChar_vdu40(int charnum, int x, int y, bitmap_ind16 &bitmap) const;
+	void decode_videoram_char(int row, int col, uint8_t &chr, uint8_t &attrib) const;
```

Comment:
Marking member functions as `const` when they actually change state is generally frowned upon, even when you can technically do it because everything is modified via pointers.  It helps the *developer* as well as the compiler if you think about semantics.

Reply:
These particular functions don't change state in my mind.  For example putChar_vdu80 draws a character onto a bitmap that is passed in by reference.  Nothing in the object itself changes, and the bitmap is part of the display rather then the machine itself.  Or am I thinking about it the wrong way?  I don't really mind if they are const, but in my day job I would naturally make them const so it's hard to change mindset.

The palette functions that you highlight in another comment are different.  I can see that these are conceptually not const as the palette is stored inside the machine.